### PR TITLE
Users

### PR DESCRIPTION
**Pull request by Synkatathesis**
added bcryptjs, created a model for users and requests to create update and delete users, still a work in progress but give me your thoughts on this

### CODE REVIEW

**Review by seanjhardy — CHANGES_REQUESTED** [review]
Looking awesome :D One thing I'd say is that the "badges" for a user, are equivalent to their completed skills. We can fetch the badge by their skill instead of storing it separately. Also we need completed skills and current skills. The skills/items should reference ObjectIDs to other objects in the DB. Also the default icon path doesn't work for me (also not sure if we'll even store it? but probably fine to leave it in for now)

**Comment by Synkatathesis** [comment]
> Looking awesome :D One thing I'd say is that the "badges" for a user, are equivalent to their completed skills. We can fetch the badge by their skill instead of storing it separately. Also we need completed skills and current skills. The skills/items should reference ObjectIDs to other objects in the DB. Also the default icon path doesn't work for me (also not sure if we'll even store it? but probably fine to leave it in for now)

Thanks ! I went and fixed a few things, yeah the icon path changed since the last time I used it so I changed it, should be fine now

**Comment by seanjhardy** [comment]
Yeah, but we'll also need backwards compatibility with users who don't have discord and who want to use the app. I think what we need to do is store the username/password, as well as any other integration/signin methods like discord, so both users can authenticate with whatever service they choose. Might deserve more of a discussion than github comments imo

**Review by seanjhardy — APPROVED** [review]

File: src/models/user.model.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    },
+    password: {
+        type:String,
+        required:true,
+    },
+    pic:{
+        type:String,
+        required:true,
+        default:
+            "https://icon-library.com/icon/anonymous-avatar-icon-25.html"
```
Comment (swatarianess) [inline src/models/user.model.js:28]: Replace this with <https://icon-library.com/images/anonymous-avatar-icon/anonymous-avatar-icon-25.jpg> as its a direct link to the image file. 👍🏾

**Review by swatarianess — CHANGES_REQUESTED** [review]

**Comment by Synkatathesis** [comment]
There all fixed for now, feel free to tell me if you think there's anything else wrong or missing

**Review by swatarianess — APPROVED** [review]

**Pull request merged**